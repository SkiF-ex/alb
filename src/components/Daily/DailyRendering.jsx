import React from "react";
import {useEffect, useState} from "react";
import StickerCard from "../molecules/StickerCard";
import StickerPack from "../molecules/StickerPack";
import getCounter from "../../utils/getCounter";


export const DailyRendering = ({ useGetStickers }) => {
  const [
    getStickers,
    stickerPacks,
    setStickerPacks,
    newStickerPack,
    newStickersOpened,
    setNewStickersOpened,
  ] = useGetStickers();
  const [mock, setMock] = useState([]);

  useEffect(() => {
    getCounter(setStickerPacks, setMock);
  }, [])

  const stickerPacksTree = () => {
    return (
      <div className="stickers_window">
        <p className="stickers_header_title">Daily sticker sets</p>
        <p className="stickers_header_subtitle">you have {stickerPacks} sticker sets to open</p>
        <div className="stickers_block">
          {mock.map((elem) => <StickerPack id={elem} handleOpen={getStickers}/>)}
        </div>
      </div>)
  }

  const stickerSetTree = () => {
    return (
      <div className="stickers_window">
        <p className="stickers_header_title">Sticker set has opened</p>
        <p className="stickers_header_subtitle">you have 6 new stickers</p>
        <div className="stickers_block">
          {newStickerPack.map((elem) => <StickerCard sticker={elem}/>)}
        </div>
        <button className="button" onClick={() => {
          setNewStickersOpened(false);
          getCounter(setStickerPacks, setMock);
        }}>Paste
        </button>
      </div>)
  }

  return (
    <>
      {newStickersOpened ? stickerSetTree() : stickerPacksTree()}
    </>
  )
}
