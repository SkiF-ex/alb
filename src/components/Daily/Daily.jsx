import React from "react";

import { useGetStickers } from './hooks/getStickers';
import { DailyRendering } from './DailyRendering';

const Daily = () => {
    return (
      <DailyRendering useGetStickers={useGetStickers} />
    )
}

export default Daily;
