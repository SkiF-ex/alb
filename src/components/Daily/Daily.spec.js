import Daily from "./Daily";
// import {shallow} from "enzyme";
import { act, create } from 'react-test-renderer';



import React from "react";

// const component = shallow(<Daily />);

describe('Daily component', () => {

  it('should render component', () => {
    let component;
    act(() => {
      component = create(
        <Daily />
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
    // const wrapper = component.find(".stickers_window");
    // expect(wrapper.length).toBe(1);
  });
})
