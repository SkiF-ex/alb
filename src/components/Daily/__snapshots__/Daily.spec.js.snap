// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Daily component should render component 1`] = `
<div
  className="stickers_window"
>
  <p
    className="stickers_header_title"
  >
    Daily sticker sets
  </p>
  <p
    className="stickers_header_subtitle"
  >
    you have 
    
     sticker sets to open
  </p>
  <div
    className="stickers_block"
  />
</div>
`;
